File: Design/postEnergyData.py
import requests
import time
import sys
import json

def postEnergyData():
    orgName = sys.argv[1]
    if (orgName == 'org1'):
        url = 'http://localhost:10000/postEnergyData'
    else :
        url = 'http://localhost:10001/postEnergyData'
    file = open("diploma.energy.data." + orgName + ".csv", "r")
    org = file.readline().split(': ')[1]
    frequency = file.readline().split(': ')[1].split(' ')[0]
    file.readline()

    for line in file.readlines():
        timestamp, energy = line.split('\n')[0].split(', ')
        payload = {
            "timestamp": str(timestamp),
            "energy" : str(energy)
        }
        x = requests.post(url, json = payload)
        print(x.text)
        #sleep for 50 miliseconds
        # time.sleep(.05)                             

    file.close()

if (len(sys.argv[1:]) != 1):
    print("Error: please give organization name as argument")
elif (sys.argv[1] == 'org1' or sys.argv[1] == 'org2'):
    postEnergyData()
else:
    print("Error: please give 'org1' or 'org2'")

File: Design/postEnergyData.js
fs = require('fs')
fetch = require('node-fetch')

let args = []
let url = ''
sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms))

postEnergyData = async () => {
    let orgName = args[0]
    if (orgName == 'org1') 
        url = 'http://localhost:10000/postEnergyData'
    else
        url = 'http://localhost:10001/postEnergyData'
    
    let fileName = 'diploma.energy.data.' + orgName + '.csv'
    fs.readFile(fileName, 'utf8', async (err,data) => {
        if (err)  
            return console.log(err)
        let lines = data.split('\n')
        lines = lines.slice(3)
        // sleep 1 second for every day. Otherwise the blockchain would block
        // us as a malicious user
        for (let day=1; day<=31; day++){
            await postEnergyDataPerDay(lines.slice(96*(day-1)+1, 96*day))
            await sleep(1000)
        }
    })
}

postEnergyDataPerDay = (lines) => {
    lines.forEach(line => {
        let [timestamp, energy] = line.split('\n')[0].split(', ')
        const requestOptions = {
            method: 'POST',
            headers: {'Content-Type': 'application/json'},
            body: JSON.stringify({
                "timestamp": timestamp.toString(),
                "energy" : energy.toString()
            })
        }
        fetch(url, requestOptions)
        .then(
            (response) => {
                if (response.status !== 200) {
                    console.log('Looks like there was a problem. Status Code: ' + response.status)
                    return
                }

                // Examine the text in the response
                response.json().then((data) => {
                    console.log(data);
                });
            }
        )
        .catch(function(err) {
            console.log('Fetch Error :-S', err);
        })
    })
}

args = process.argv.slice(2)
if (args.length != 1)
    console.log("Error: please give organization name as argument")
else if (args[0] == 'org1' || args[0] == 'org2')
    postEnergyData()
else
    console.log("Error: please give 'org1' or 'org2'")

